### DISCUSSION TITLE
Going through the source code and matching it with the research paper?

### DISCUSSION THREAD

**Discussion by amaarora** [rwightman/efficientdet-pytorch#160] [Q&A]
Anybody interested in going through the source code and matching it with the research paper to understand how `EfficientDet` is implemented in PyTorch?

**Comment by rwightman** [comment]
@amaarora for you or anyone who decides to join you doing that, worth mentioning that a lot of people were confused by the initial paper when trying to replicate the models. All attempts to create the model before the official Tensorflow version was released didn't quite get it right... especially with details of the P6/P7 features. 

I implemented this with the TF imp as guidance (as of March 2020, it has changed quite a bit since then with a Keras re-write).  I did not pay as much attention to the paper. https://github.com/google/automl/tree/master/efficientdet

**Reply by amaarora** [reply]
Thanks @rwightman ! I have actually been able to understand your implementation using the official TF implementation step-by-step. 

I am working on writing a blog post - something like "The Annotated EfficientDet" that shows how to create `EfficientDet` using your implementation by matching the implementation with the research paper. Should be done by Monday 11 Jan. :)

I quite enjoyed reading the source code - it was mostly easy to follow. Thanks for your hard work :)
